W-17802301 kafka connector improve gaps (#2812)
mulesoft/docs-connectors: kafka/4.10/modules/ROOT/pages/kafka-connector-reference.adoc

| Scope a| String |  Defines the specific permissions and access rights granted to the client application by the authorization server. |  | 
| Audience Key a| String |  Specifies the configuration key used to define the OAuth audience within the JAAS configuration. |  audience | 
| Audience a| String |  Comma separated list of one or more target audiences for the access token. |  | 

> | Max Token Expiration Seconds a| Number |  Sets the maximum duration, in seconds, for which an access token remains valid. This helps prevent unauthorized access by limiting the access token's lifespan, ensuring that even if the access token is intercepted, it will become invalid after a set period, thus enhancing security and enforcing re-authentication. |  | 

Reviewer: ```suggestion
| Max Token Expiration Seconds a| Number |  Sets the maximum duration, in seconds, for which an access token remains valid. This helps prevent unauthorized access by limiting the access token's lifespan, ensuring that even if the access token is intercepted, it becomes invalid after a set period, thus enhancing security and enforcing re-authentication. |  | 
```

| Principal Name a| String |  Unique identifier associated with an authenticated user, often used for authorization decisions and resource access. |  | 
| Credentials Placement a| Enumeration, one of:

** BASIC_AUTHENTICATION_HEADER
** BODY |  Defines whether to include client credentials in the basic authentication header or in the body of the authentication request. |  BASIC_AUTHENTICATION_HEADER | 
| OAuth Module Required a| Boolean |  Indicates whether this login module must be successfully authenticated for Kafka clients to establish a connection using OAuth 2.0 bearer tokens. |  true | 
| Include Accept Header a| Boolean |  Whether to include 'Accept-application/json' header in the authentication request. |  true | 

> | OAuth Extensions a| Object |  Represents a mapping of key-value pairs containing extensions for custom broker OAuth implementations. If you specify a key for the OAuth Extensions field, ensure that the input contains only alphabetic characters and is at least one character long. If you specify a value for OAuth Extensions field, ensure the input contains only printable ASCII characters or whitespace (including space, tab, carriage return, and newline), and the value is at least one character long. You do not need to include the `extension_` part, such as in `extension_logicalCluster` or `extension_identityPoolId`. |  | 

Reviewer: ```suggestion
| OAuth Extensions a| Object |  Represents a mapping of key-value pairs containing extensions for custom broker OAuth implementations. If you specify a key for the OAuth Extensions field, ensure that the input contains only alphabetic characters and is at least one character long. If you specify a value for OAuth Extensions field, ensure the input contains only printable ASCII characters or whitespace (including space, tab, carriage return, and newline), and the value is at least one character long. You don't need to include the `extension_` part, such as in `extension_logicalCluster` or `extension_identityPoolId`. |  | 
```

Reviewer: Maybe you can also say: "Don't include the ..." to avoid "need to".

| Scope a| String |  Defines the specific permissions and access rights granted to the client application by the authorization server. |  | 
| Audience Key a| String |  Specifies the configuration key used to define the OAuth audience within the JAAS configuration. |  audience | 
| Audience a| String |  Comma separated list of one or more target audiences for the access token. |  | 

> | Max Token Expiration Seconds a| Number |  Sets the maximum duration, in seconds, for which an access token remains valid. This helps prevent unauthorized access by limiting the access token's lifespan, ensuring that even if the access token is intercepted, it will become invalid after a set period, thus enhancing security and enforcing re-authentication. |  | 

Reviewer: ```suggestion
| Max Token Expiration Seconds a| Number |  Sets the maximum duration, in seconds, for which an access token remains valid. This helps prevent unauthorized access by limiting the access token's lifespan, ensuring that even if the access token is intercepted, it becomes invalid after a set period, thus enhancing security and enforcing re-authentication. |  | 
```

| Principal Name a| String |  Unique identifier associated with an authenticated user, often used for authorization decisions and resource access. |  | 
| Credentials Placement a| Enumeration, one of:

** BASIC_AUTHENTICATION_HEADER
** BODY |  Defines whether to include client credentials in the basic authentication header or in the body of the authentication request. |  BASIC_AUTHENTICATION_HEADER | 
| OAuth Module Required a| Boolean |  Indicates whether this login module must be successfully authenticated for Kafka clients to establish a connection using OAuth 2.0 bearer tokens. |  true | 
| Include Accept Header a| Boolean |  Whether to include 'Accept-application/json' header in the authentication request. |  true | 

> | OAuth Extensions a| Object |  Represents a mapping of key-value pairs containing extensions for custom broker OAuth implementations. If you specify a key for the OAuth Extensions field, ensure that the input contains only alphabetic characters and is at least one character long. If you specify a value for OAuth Extensions field, ensure the input contains only printable ASCII characters or whitespace (including space, tab, carriage return, and newline), and the value is at least one character long. You do not need to include the `extension_` part, such as in `extension_logicalCluster` or `extension_identityPoolId`. |  | 

Reviewer: ```suggestion
| OAuth Extensions a| Object |  Represents a mapping of key-value pairs containing extensions for custom broker OAuth implementations. If you specify a key for the OAuth Extensions field, ensure that the input contains only alphabetic characters and is at least one character long. If you specify a value for OAuth Extensions field, ensure the input contains only printable ASCII characters or whitespace (including space, tab, carriage return, and newline), and the value is at least one character long. You don't need to include the `extension_` part, such as in `extension_logicalCluster` or `extension_identityPoolId`. |  | 
```
